### PR TITLE
[DMS-30] Update GitHub actions

### PR DESCRIPTION
**Pull request by stephenfuqua**
Substantial updates to Node versions and GitHub actions, and fixes to the PowerShell linter.

To test, I temporarily modified my fork of the DMS project to use my fork of Ed-Fi-Actions. [In this test run](https://github.com/stephenfuqua/Data-Management-Service/actions/runs/8176393746), you can see that there are no warnings or failures from the shared actions. There are unrelated warnings/failures in the DMS fork, which can be ignored in this context.

The pull request jobs _for this repository itself_ have now also [been tested in my fork](https://github.com/stephenfuqua/Ed-Fi-Actions/actions): "pull request" and "after pull request" jobs are green.

### CODE REVIEW

File: .github/workflows/after-pullrequest-powershell.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Upload Test Results
-        uses: dorny/test-reporter@c9b3d0e2bd2a4e96aaf424dbaa31c46b42318226 #v1.6.0
+        uses: dorny/test-reporter@eaa763f6ffc21c7a37837f56cd5f9737f27fc6c8 # v1.8.0
```
[the remark this answers is not in the document]
Comment (andonyns) [reply .github/workflows/after-pullrequest-powershell.yml:19]: This happens when we upgrade upload-artifacts

[the remark this answers is not in the document]
Comment (stephenfuqua) [reply .github/workflows/after-pullrequest-powershell.yml:19]: I'm testing out a fork of dorny/test-reporter now.

File: .github/workflows/after-pullrequest-bidi.yml [hunk 2]
```diff
@@ -13,13 +13,28 @@ on:
 
 jobs:
   report-test-results:
+    name: Report Test Results
     runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion != 'skipped'
```
Comment (andonyns) [inline .github/workflows/after-pullrequest-bidi.yml:18]: I like this approach

**Review by andonyns — CHANGES_REQUESTED** [review]

File: bidi-scanner/action.yml [hunk 3]
```diff
@@ -25,5 +25,5 @@ inputs:
     default: ''
 # outputs: Build failure if characters are detected.
 runs:
-  using: node16
+  using: node20
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply bidi-scanner/action.yml:28]: `node18` does not exist in GitHub, in this context.

File: action-allowedlist/README.md [hunk 4]
```diff
@@ -46,12 +46,14 @@ jobs:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
       - uses: ./action-allowedlist
         name: Scan used actions
         id: scan-action
+
       - name: Upload result file as artifact
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
+        uses: actions/26f96dfa697d77e81fd5907df203aa23a56210a8 # v4.3.0
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply action-allowedlist/README.md:56]:
```suggestion
        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 # v4.3.0
```

File: bidi-scanner/github/package.json [hunk 5]
```diff
@@ -2,9 +2,9 @@
   "name": "@edfi/bidi-scanner-action",
   "type": "module",
   "engines": {
-    "node": ">=16.0.0"
+    "node": ">=18.0.0"
```
Comment (stephenfuqua) [inline bidi-scanner/github/package.json:5]: Node 18 _does_ exist as an option when running custom code. You just can't choose `node18` in `action.yml`.

**Review by andonyns — APPROVED** [review]

**Pull request merged**